sdk/core/azure-core/test/libcurl-stress-test/Dockerfile

```diff
@@ -3,11 +3,20 @@
 
 FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04
 
-# copy the tagrget binary
-COPY ./bin/libcurl-stress-test/azure-core-libcurl-stress-test ./azure-core-libcurl-stress-test
-RUN chmod +x ./azure-core-libcurl-stress-test
-
-# install the mem check tool 
+# install the mem check tool along side the other deps
 RUN apt-get update -y
-RUN apt-get install valgrind -y
+RUN apt-get install gcc cmake make g++ git zip unzip build-essential pkg-config wget curl valgrind -y
+RUN wget -O vcpkg.tar.gz https://github.com/microsoft/vcpkg/archive/master.tar.gz
+RUN mkdir /opt/vcpkg
+RUN tar xf vcpkg.tar.gz --strip-components=1 -C /opt/vcpkg
+RUN /opt/vcpkg/bootstrap-vcpkg.sh
+RUN ln -s /opt/vcpkg/vcpkg /usr/local/bin/vcpkg
 
+RUN git clone https://github.com/Azure/azure-sdk-for-cpp.git 
+WORKDIR /azure-sdk-for-cpp/build
+RUN cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON -DBUILD_TRANSPORT_CURL=ON ..
+RUN cmake --build . --target azure-core-libcurl-stress-test 
+
```

Comment:
I think we can split this Dockerfile here and have the first part be a build container and the second part be the actual final container. This lets you create a runtime image that doesn't have all the dev tools and source code.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/internal/stress/Dockerfile

This is the start of the build container:
https://github.com/Azure/azure-sdk-for-go/blob/552351111cd8a8ec9a7619e0eacf644e1dbe5905/sdk/messaging/azservicebus/internal/stress/Dockerfile#L1

# this is the runtime container
https://github.com/Azure/azure-sdk-for-go/blob/552351111cd8a8ec9a7619e0eacf644e1dbe5905/sdk/messaging/azservicebus/internal/stress/Dockerfile#L15

Reply:
@gearama here is a working dockerfile that will build from whatever version the checked out repo is on (working both locally and in CI). However it takes ~5 mins to build on my CPU hungry machine, is this expected?

NOTE: in order for this to work, the `imageBuildDir` field in the scenarios-matrix must be updated:

```
imageBuildDir: "../../../../../"
```

Dockerfile:

```
# Copyright (c) Microsoft Corporation. All rights reserved.
# SPDX-License-Identifier: MIT

FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04 as build
# FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04

# install the mem check tool along side the other deps
RUN apt-get update -y
RUN apt-get install -y gcc cmake make g++ git zip unzip build-essential pkg-config wget curl valgrind
RUN wget -O vcpkg.tar.gz https://github.com/microsoft/vcpkg/archive/master.tar.gz
RUN mkdir /opt/vcpkg
RUN tar xf vcpkg.tar.gz --strip-components=1 -C /opt/vcpkg
RUN /opt/vcpkg/bootstrap-vcpkg.sh
RUN ln -s /opt/vcpkg/vcpkg /usr/local/bin/vcpkg

ADD . /src
WORKDIR /build
RUN cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON -DBUILD_TRANSPORT_CURL=ON /src
RUN cmake --build . --target azure-core-libcurl-stress-test

FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04

RUN apt-get update -y
RUN apt-get install -y valgrind
WORKDIR /

# copy the target binary
COPY --from=build ./build/sdk/core/azure-core/test/libcurl-stress-test/azure-core-libcurl-stress-test ./azure-core-libcurl-stress-test
RUN chmod +x ./azure-core-libcurl-stress-test

CMD valgrind --tool=memcheck -s ./azure-core-libcurl-stress-test
```

Reply:
i trust your code is correct , i got it things working local , only that when trying to use deploy.ps1 to deploy in the test cluster i keep getting event for the pod : 


 Warning   FailedScheduling   pod/cpp-constantdetach-libcurl-stress-test-9-v4sb6   0/8 nodes are available: 2 node(s) had untolerated taint {CriticalAddonsOnly: true}, 3 node(s) didn't match Pod's node affinity/selector, 6 Insufficient cpu. preemption: 0/8 nodes are available: 3 Preemption is not helpful for scheduling, 5 No preemption victims found for incoming pod.

Reply:
@gearama you might have been hitting noisy neighbors on the cluster with node usage. Looks like the cluster scaled up node count in response and was able to schedule your test. I bumped up the node count baseline as well.